cmd/jujud/agent/machine.go

```diff
@@ -1198,8 +1203,12 @@ func (a *MachineAgent) apiserverWorkerStarter(
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		w, err := a.newAPIserverWorker(st, certChanged, dependencyReporter)
+		statePool := state.NewStatePool(st)
+		w, err := a.newAPIserverWorker(
+			st, statePool, certChanged, dependencyReporter,
+		)
 		if err != nil {
+			statePool.Close()
```

Comment:
StatePool.Close closes the non-system States in the pool. The system state (i.e. the one passed into NewStatePool) must be managed separately.